pkg/cmd/server/api/validation/master.go

```diff
@@ -61,6 +61,13 @@ func ValidateMasterConfig(config *api.MasterConfig) ValidationResults {
 		validationResults.AddErrors(urlErrs...)
 	}
 
+	switch {
+	case config.ControllerLeaseTTL > 300:
+		validationResults.AddErrors(fielderrors.NewFieldInvalid("controllerLeaseTTL", config.ControllerLeaseTTL, "TTL must be -1 (disabled), 0 (default), or between 10 and 300 seconds"))
+	case config.ControllerLeaseTTL > 0 && config.ControllerLeaseTTL < 10:
+		validationResults.AddErrors(fielderrors.NewFieldInvalid("controllerLeaseTTL", config.ControllerLeaseTTL, "TTL must be -1 (disabled), 0 (default), or between 10 and 300 seconds"))
+	}
```

Comment:
I've done a lot of reviews today, but I'm pretty sure this code allows `-100` in conflict with your messages